packages/core/src/mixer.ts

```diff
@@ -3,18 +3,16 @@ import type HoprCoreConnector from '@hoprnet/hopr-core-connector-interface'
 import Heap from 'heap-js'
 import { randomInteger } from '@hoprnet/hopr-utils'
 import { MAX_PACKET_DELAY } from './constants'
-import debug from 'debug'
-const log = debug('hopr-core:mixer')
+import Defer from 'p-defer'
+import type { DeferredPromise } from 'p-defer'
+
+import Debug from 'debug'
+const log = Debug('hopr-core:mixer')
 
 type HeapElement = [number, Packet<any>]
 
 const comparator = (a: HeapElement, b: HeapElement): number => {
-  if (b[0] < a[0]) {
-    return 1
-  } else if (b[0] > a[0]) {
-    return -1
-  }
-  return 0
+  return a[0] - b[0]
```

Comment:
This is incorrect - the comparator needs to return -1, 0 or 1 as before.

Reply:
done